### PR TITLE
Cartopy updates

### PR DESCRIPTION
**Pull request by jrleeman**
Update to use the easier state border access method available in carroty >=0.16 and very minor cleanup surrounding those lines.

### CODE REVIEW

File: notebooks/Satellite_Data/GOES_Interactive_Plot.ipynb [hunk 1]
```diff
@@ -19,7 +19,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 1,
```
Comment (dopplershift) [inline notebooks/Satellite_Data/GOES_Interactive_Plot.ipynb]: Shouldn't we be clearing all output before committing?

**Comment by dopplershift** [comment]
Did you look for anywhere else that `with_scale` could be used?

File: notebooks/Satellite_Data/GOES_Interactive_Plot.ipynb (hunk 1 above)
Comment (jrleeman) [reply notebooks/Satellite_Data/GOES_Interactive_Plot.ipynb]: I've been careful on the exercise notebooks. This is a demo notebook so I could go either way. Web rendering = good

**Comment by jrleeman** [comment]
I added it on the small scale stuff. We probably need to do a complete run over of each notebook soon for MetPy/siphon 0.8

**Pull request merged**